test.py

```diff
@@ -253,10 +253,20 @@ def test(data,
     # Compute statistics
     stats = [np.concatenate(x, 0) for x in zip(*stats)]  # to numpy
     if len(stats) and stats[0].any():
-        p, r, ap, f1, ap_class = ap_per_class(*stats, plot=plots, fname=save_dir / 'precision-recall_curve.png')
+        p, r, ap, f1, ap_class, eer_precision, eer_recall, threshold = ap_per_class(*stats, plot=plots, fname=save_dir / 'precision-recall_curve.png')
         p, r, ap50, ap = p[:, 0], r[:, 0], ap[:, 0], ap.mean(1)  # [P, R, AP@0.5, AP@0.5:0.95]
         mp, mr, map50, map = p.mean(), r.mean(), ap50.mean(), ap.mean()
         nt = np.bincount(stats[3].astype(np.int64), minlength=nc)  # number of targets per class
+        
+        #output to json for dvc metrics
+        data = {"equal_error_rate":{"precision":eer_precision.mean(),
+                                    "recall":eer_recall.mean(),
+                                    "threshold":threshold.mean()},
+                "average_precision":map50}
+        filename = 'metrics/dev/metrics.json'
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
```

Comment:
I prefer to use the pathlib.Path API where possible.
`filename.parent.mkdir(exist_ok=True, parents=True)`